privacy-notice.js

```diff
@@ -0,0 +1,110 @@
+// Creates a cookie to indicate acknowledgment of privacy notice
+var setCookie = function() {
+  var now = new Date();
+  var expiration = new Date(now.setFullYear(now.getFullYear() + 1));
+  document.cookie = "mitlibPrivAck=true;domain=.mit.edu;path=/;expires=" + expiration;
+
+  // Remove the privacy notice once the cookie is set
+  var privacyNode = document.getElementById('privacy-notice');
+  if (privacyNode.parentNode) {
+    privacyNode.parentNode.removeChild(privacyNode);
+  }
+}
+
+// Fetch privacy notice acknowledgment cookie
+var getCookieValue = function(cookieName) {
+  // Go no further if cookies are empty
+  if (!document.cookie) {
+    return null;
+  }
+
+  // Split document.cookie into an array
+  var cookies = document.cookie.split(';');
+
+  // Initialize this to null so it returns falsey if the value isn't found
+  var match = null;
+
+  // Look for the cookie name in the array
+  var i;
+  
+  for (i = 0; i < cookies.length; i++) {
+    var cookiePair = cookies[i].split('=');
+
+    if (cookieName == cookiePair[0].trim()) {
+      match = decodeURIComponent(cookiePair[1]);
+    }   
+  }
+
+  // Return the cookie value if found (or null otherwise)
+  return match;
+}
+
+// https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule
+var addStylesheetRules = function(rules) {
+  var styleEl = document.createElement('style');
+
+  // Append <style> element to <head>
+  document.head.appendChild(styleEl);
+
+  // Grab style element's sheet
+  var styleSheet = styleEl.sheet;
+
+  for (var i = 0; i < rules.length; i++) {
+    var j = 1, 
+        rule = rules[i], 
+        selector = rule[0], 
+        propStr = '';
+    // If the second argument of a rule is an array of arrays, correct our variables.
+    if (Array.isArray(rule[1][0])) {
+      rule = rule[1];
+      j = 0;
+    }
+
+    for (var pl = rule.length; j < pl; j++) {
+      var prop = rule[j];
+      propStr += prop[0] + ': ' + prop[1] + (prop[2] ? ' !important' : '') + ';\n';
+    }
+
+    // Insert CSS Rule
+    styleSheet.insertRule(selector + '{' + propStr + '}', styleSheet.cssRules.length);
+  }
+}
+
+// Display privacy notice if the acknowledgment cookie isn't set
+var privacyNotice = function() {
+  if (!getCookieValue('mitlibPrivAck')) {
```

Comment:
If we could have one place at the top of the file to set the currently active cookie name and read it both here and on line 5 I think we might save ourselves some future confusion when we forget and only update one.